### PR TITLE
fixed search structure for Book class

### PR DESCRIPTION
**Pull request by yuriy-kononchuk**

### CODE REVIEW

**Review by Ivan95kos — APPROVED** [review]
Great job

File: src/main/java/com/example/project/exception/CustomGlobalExceptionHandler.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    private String getErrorMessage(ObjectError error) {
+        if (error instanceof FieldError) {
+            String field = ((FieldError) error).getField();
+            String message = error.getDefaultMessage();
+            return field + " " + message;
+        }
+        return error.getDefaultMessage();
+    }
+
+    protected ResponseEntity<Object> handleEntityNotFoundException(
```
Comment (Ivan95kos) [inline src/main/java/com/example/project/exception/CustomGlobalExceptionHandler.java:46]: You can add the `@ExceptionHandler(value = {EntityNotFoundException.class})`

File: src/main/java/com/example/project/dto/CreateBookRequestDto.java [hunk 2]
```diff
@@ -1,15 +1,22 @@
...
 import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import jakarta.validation.constraints.Min;
+import jakarta.validation.constraints.NotNull;
 import java.math.BigDecimal;
 import lombok.Data;
 
 @Data
 @JsonIgnoreProperties(ignoreUnknown = true)
 public class CreateBookRequestDto {
+    @NotNull
```
Comment (Ivan95kos) [inline src/main/java/com/example/project/dto/CreateBookRequestDto.java:12]: It is better to use the @NotEmpty or @NotBlank for Strings
https://www.baeldung.com/java-bean-validation-not-null-empty-blank

**Review by Ivan95kos — APPROVED** [review]
Great job, with minor comments

File: src/main/java/com/example/project/exception/CustomGlobalExceptionHandler.java (hunk 1 above)
Comment (yuriy-kononchuk) [reply src/main/java/com/example/project/exception/CustomGlobalExceptionHandler.java:46]: great. Thanks

File: src/main/java/com/example/project/dto/CreateBookRequestDto.java (hunk 2 above)
Comment (yuriy-kononchuk) [reply src/main/java/com/example/project/dto/CreateBookRequestDto.java:12]: great advice

**Pull request merged**